pandas/io/json/arrow_json_parser_wrapper.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pandas._typing import ReadBuffer
+from pandas.compat._optional import import_optional_dependency
+
+if TYPE_CHECKING:
+    from pandas import DataFrame
+
+
+class ArrowJsonParserWrapper:
+    """
+    Wrapper for the pyarrow engine for read_json()
+    """
+
+    def __init__(self, src: ReadBuffer[bytes]) -> None:
+        self.src = src
+
+    def read(self) -> DataFrame:
+        """
+        Reads the contents of a JSON file into a DataFrame and
+        processes it according to the kwargs passed in the
+        constructor.
+
+        Returns
+        -------
+        DataFrame
+            The DataFrame created from the JSON file.
+        """
+        pyarrow_json = import_optional_dependency("pyarrow.json")
```

Comment:
Doesn't seem like this class is doing too much. Can this all be inlined?

Reply:
When I was writing that, I thought maybe in the future if pyarrow supports more features, it could be extended. I also don't know what `Can this all be inlined?` means, can you please explain?

Reply:
Sorry, I had meant can the code here just be written directly where it's used.

We can work on refactoring this out in the future if it becomes more complex

Reply:
Oh, I think I did some if with my last commit. I figured out the `_make_engine` method was totally unnecessary so I removed it. I also got what you are saying, I will try to tackle that as well.

Reply:
@mroeschke I think I did what you asked for (refactoring to inline); however, please take a look and let me know.